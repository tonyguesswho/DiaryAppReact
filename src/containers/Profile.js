import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { signout } from '../actions/authAction';

class Profile extends Component {
  constructor() {
    super();
    this.state = {
      profile: {}
    };
  }
  logout = () => {
    signout();
  };
  componentDidMount = () => {
    const { profile } = this.props;
    this.setState({
      profile
    });
  };
  render() {
    const { profile } = this.state;
    return (
      <div className="bg body">
        <header className="header2">
          <nav>
            <ul className="nav nav2">
              <li>
                <Link to="/">Home</Link>
              </li>
              <li>
                <Link to="/" id="logout" onClick={this.logout}>
                  Sign Out
                </Link>
              </li>
            </ul>
          </nav>
        </header>
        <section className="profile">
          <div className="bigbox">
            <div className="box card span21 span2-center">
              <h2 id="username">Welcome {profile.username}</h2>

              <Link to="/create" className="ebtn ebtn-filled">
                Add Entry
              </Link>
              <Link to="/entries" className="ebtn ebtn-transparent">
                My Diary
              </Link>
            </div>
          </div>
        </section>
      </div>
    );
  }
}
const mapStateToProps = state => ({ profile: state.profile.userProfile });
export default connect(
  mapStateToProps,
  { signout }
)(Profile);
