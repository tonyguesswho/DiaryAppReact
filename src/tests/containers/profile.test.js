import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import Profile from '../../containers/Profile';
import Root from '../../root';

let wrapped;

beforeEach(() => {
  wrapped = mount(
    <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
      <Root>
        <Profile />
      </Root>
    </MemoryRouter>
  );
});

afterEach(() => wrapped.unmount());

describe('Profile entry UI', () => {
  describe('render features', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});
