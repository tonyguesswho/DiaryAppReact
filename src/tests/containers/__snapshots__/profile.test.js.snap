// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile entry UI render features container should render as expected 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <_default>
      <Provider
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <Connect(Profile)>
          <Profile
            profile={Object {}}
            signout={[Function]}
          >
            <div
              className="bg body"
            >
              <header
                className="header2"
              >
                <nav>
                  <ul
                    className="nav nav2"
                  >
                    <li>
                      <Link
                        replace={false}
                        to="/"
                      >
                        <a
                          href="/"
                          onClick={[Function]}
                        >
                          Home
                        </a>
                      </Link>
                    </li>
                    <li>
                      <Link
                        id="logout"
                        onClick={[Function]}
                        replace={false}
                        to="/"
                      >
                        <a
                          href="/"
                          id="logout"
                          onClick={[Function]}
                        >
                          Sign Out
                        </a>
                      </Link>
                    </li>
                  </ul>
                </nav>
              </header>
              <section
                className="profile"
              >
                <div
                  className="bigbox"
                >
                  <div
                    className="box card span21 span2-center"
                  >
                    <h2
                      id="username"
                    >
                      Welcome 
                    </h2>
                    <Link
                      className="ebtn ebtn-filled"
                      replace={false}
                      to="/create"
                    >
                      <a
                        className="ebtn ebtn-filled"
                        href="/create"
                        onClick={[Function]}
                      >
                        Add Entry
                      </a>
                    </Link>
                    <Link
                      className="ebtn ebtn-transparent"
                      replace={false}
                      to="/entries"
                    >
                      <a
                        className="ebtn ebtn-transparent"
                        href="/entries"
                        onClick={[Function]}
                      >
                        My Diary
                      </a>
                    </Link>
                  </div>
                </div>
              </section>
            </div>
          </Profile>
        </Connect(Profile)>
      </Provider>
    </_default>
  </Router>
</MemoryRouter>
`;
